libp2p/switch.nim

```diff
@@ -471,6 +468,16 @@ proc muxerHandler(s: Switch, muxer: Muxer) {.async, gcsafe.} =
       await stream.close()
 
   try:
+    if isNil(stream.peerInfo):
+      await muxer.close()
+      return
+
+    # store incoming connection
+    s.connManager.storeIncoming(muxer.connection)
```

Comment:
I've just tested this branch and it's not finalizing with Lighthouse. I expected this, because I've tested the same `storeIncoming` location in my test branch as well. It's not just `identity` that can introduce a race, but rather the very first line here (`await muxer.newStream()`) and that's why I had to put the storeIncoming lines before any other async op here. Sorry for the misleading explanation.

Reply:
With the additional tweak here, it does finalize.

Reply:
I was hoping a little though that the de-asyncified creation of mplex streams in `simplify mplex` would have helped but the `open` op is still async so .. well ..

Reply:
It did help in the sense that I had to make several runs to trigger the problem :) A better hiding place for the bug.

Reply:
It should be fine stirring the connection early, if the mixer close because of an error it will be cleaned up from the manager.